curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/inpaint.html

```diff
@@ -0,0 +1,129 @@
+<html>
+<head>
+<meta charset="utf-8">
+<title>CVdrawing</title>
+</head>
+<body>
+<p id="status">OpenCV.js is loading...</p>
+<canvas id="canvas" width="640" height="426">
+</canvas>
+<img id="src" src="bird.jpg" style="display:none"/>
+<input type="button" id="myButton" value="Test" disabled=true
+onclick="inpaint()"/>
+
+<script type="text/javascript">
+
+var checker = false;
+var src;
+var mask;
+var dst;
+var p_x = 0;
+var p_y = 0;
+var paint;
+var srcmat;
+var maskmat;
+var dstmat;
+var x;
+var y;
+
+window.onload = function()
+{
+context = document.getElementById("canvas").getContext("2d");
+img = document.getElementById("src");
+context.drawImage(img, 0, 0);
+};
+
+document.getElementById("canvas").onmousedown = function(e)
+{
+  paint = true;
+  x = e.pageX - this.offsetLeft;
+  y = e.pageY - this.offsetTop;
```

Comment:
Can we achieve the same using just 
```
x = e.x;
y = e.y;
```
?

source: https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/x

Reply:
It seems to me that we can store `p_x` and `p_y` only but get `x` and `y` in the `onmousemove`.

Reply:
https://stackoverflow.com/questions/6073505/what-is-the-difference-between-screenx-y-clientx-y-and-pagex-y
We need exactly e.pageX and e.pageY in our sample.

Reply:
got it, thanks!